### PR TITLE
[wait for #1350]  [batchnorm] Update to LayerV2 

### PR DESCRIPTION
**Pull request by kparichay**
Update batch norm to layer v2 style.
Add corresponding common unittests and enable modelfile
and models unittest.

Signed-off-by: Parichay Kapoor <pk.kapoor@samsung.com>

### CODE REVIEW

File: nntrainer/layers/bn_layer.cpp [hunk 1]
```diff
@@ -32,117 +31,127 @@
...
-  if (getNumInputs() != 1) {
+void BatchNormalizationLayer::finalize(InitLayerContext &context) {
+  if (context.getNumInputs() != 1) {
     throw std::invalid_argument(
       "Only one input is allowed for batch normalization layer");
   }
 
-  output_dim[0] = input_dim[0];
+  std::vector<TensorDim> output_dims(1);
+
```
Comment (jijoongmoon) [inline nntrainer/layers/bn_layer.cpp:46]: is it used?

**Review by jijoongmoon — APPROVED** [review]
LGTM with minor comments.

File: nntrainer/layers/bn_layer.cpp (hunk 1 above)
Comment (kparichay) [reply nntrainer/layers/bn_layer.cpp:46]: my bad, will be removed.

**Review by zhoonit — APPROVED** [review]
LGTM
